src/modules/textFacets/__tests__/textFacets.test.js

```diff
@@ -0,0 +1,152 @@
+            <div className="-count">({count})</div>
+        </div>
+    );
+};
+
+// establish API mocking before all tests
+beforeAll(() => {
+    window.fetch = jest.fn(() => {
+        return Promise.resolve({
+            json: () => Promise.resolve(searchResponse),
+        });
+    });
+})
+
+const attributesMap = {
+    productName: 'title',
+    uniqueId: 'uniqueId',
+    imageUrl: 'imageUrl',
+    price: 'min_cheapest_default_price',
+    productUrl: 'productUrl'
+};
+
+test('Match Snapshot for text facets', () => {
+    const tree = renderer
+        .create(
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >
+                <Products attributesMap={attributesMap} />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+
+        )
+        .toJSON();
+    expect(tree).toMatchSnapshot();
+});
+
+test('Test text facet click', async () => {
+    const { getByText } = render(
+        <>
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >   
+                <TextFacets />
+                <Products
+                    attributesMap={attributesMap}
+                />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa - 18')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa - 18"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
+        fireEvent.click(getByText("Clear"));
+    });
+});
+
+test('Test text facet click with FacetItemComponent', async () => {
+    const { getByText } = render(
+        <>
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >   
+                <TextFacets facetItemComponent={<FacetItemComponent />}/>
+                <Products
+                    attributesMap={attributesMap}
+                />
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
+        fireEvent.click(getByText("Clear"));
+    });
+});
+
+test('Test text facet click with renderProps', async () => {
```

Comment:
How is this render prop way of impl?

Reply:
Please rename the test case if it is not using render props or comment it out if it is not yet implemented completely

Reply:
yes will update it.

Reply:
done